src/api/procedures/pushBenefit.ts

```diff
@@ -0,0 +1,118 @@
+import { DividendDistribution, Identity, PolymeshError, Procedure } from '~/internal';
+import { ErrorCode, RoleType, TxTags } from '~/types';
+import { ProcedureAuthorization } from '~/types/internal';
+import { tuple } from '~/types/utils';
+import {
+  boolToBoolean,
+  corporateActionIdentifierToCaId,
+  identityIdToString,
+  stringToIdentityId,
+} from '~/utils/conversion';
+
+export interface PushBenefitParams {
+  targets: (string | Identity)[];
+}
+
+/**
+ * @hidden
+ */
+export type Params = { distribution: DividendDistribution } & PushBenefitParams;
+
+/**
+ * @hidden
+ */
+export async function preparePushBenefit(
```

Comment:
Should also throw an error if some of the identities aren't targets of the distribution